### PR TITLE
Bundler: Exclude the project itself from the list of packages

### PR DESCRIPTION
**Pull request by haikoschol**
As discussed in the previous PR addressing this special case, excluding the project is the preferrable approach. Luckily it turned out to be easier than I expected.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/529)

### CODE REVIEW

File: analyzer/src/funTest/assets/projects/synthetic/bundler/gemspec/Gemfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+source 'https://rubygems.org'
+
+gemspec
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/bundler/gemspec/Gemfile]: Missing linebreak at end of file.

Comment (haikoschol) [reply analyzer/src/funTest/assets/projects/synthetic/bundler/gemspec/Gemfile]: done

[the remark this answers is not in the document]
Comment (haikoschol) [reply analyzer/src/funTest/assets/projects/synthetic/bundler/gemspec/Gemfile]: done

File: analyzer/src/funTest/assets/projects/synthetic/bundler/gemspec/test-project.gemspec [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  gem.name        = 'test-project'
+  gem.version     = '1.0.0'
+  gem.files       = ['Gemfile']
+  gem.authors     = ['Bob Example']
+  gem.summary     = 'Test project with gemspec'
+
+  gem.add_runtime_dependency 'rack', '~>1.1'
+  gem.add_development_dependency 'rspec'
+end
+
```
[the remark this answers is not in the document]
Comment (haikoschol) [reply analyzer/src/funTest/assets/projects/synthetic/bundler/gemspec/test-project.gemspec]: done

File: analyzer/src/funTest/assets/projects/synthetic/bundler/gemspec/Gemfile.lock [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    rspec-expectations (3.7.0)
+      diff-lcs (>= 1.2.0, < 2.0)
+      rspec-support (~> 3.7.0)
+    rspec-mocks (3.7.0)
+      diff-lcs (>= 1.2.0, < 2.0)
+      rspec-support (~> 3.7.0)
+    rspec-support (3.7.1)
+
+PLATFORMS
+  ruby
```
[the remark this answers is not in the document]
Comment (haikoschol) [reply analyzer/src/funTest/assets/projects/synthetic/bundler/gemspec/Gemfile.lock]: done

File: analyzer/src/main/kotlin/managers/Bundler.kt [hunk 4]
```diff
@@ -151,61 +147,59 @@ class Bundler : PackageManager() {
...
+            // for it, and the Gemfile refers to it). In that case, skip querying Rubygems and adding Package and
+            // PackageReference objects and continue with the projects dependencies.
+            if (gemId == projectId) {
+                gemSpec.runtimeDependencies.forEach {
+                    parseDependency(workingDir, projectId, it, packages, scopeDependencies, errors)
+                }
+            } else {
+                gemSpec = queryRubygems(gemId.name, gemId.version).let {
+                    it?.merge(gemSpec) ?: gemSpec
+                }
```
[the remark this answers is not in the document]
Comment (haikoschol) [reply analyzer/src/main/kotlin/managers/Bundler.kt]: done

**Pull request merged**